src/Microsoft.AspNetCore.Razor.VSCode/src/RazorDocumentSynchronizer.ts

```diff
@@ -24,164 +25,257 @@ export class RazorDocumentSynchronizer {
     }
```

Comment:
On the preceding line, it seems odd that we register for events both in the constructor and in a public `register` method. Would it make sense to put both registrations in the same place (either the constructor, or in `register`)? Or is the distinction between the two necessary for some functionality?

Reply:
Ya I can definitely do that